nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -439,6 +439,10 @@ def _check_param_groups_mismatch(self, checkpoint_path: Union[str, Path], sharde
             bool: True if the number of param groups does not match
         """
         common_state_dict = dist_checkpointing.load_common_state_dict(checkpoint_path)
+        # @akoumparouli: if it contains an mcore dist opt, param_groups is under ['optimizer'].data['param_groups']
```

Comment:
Maybe we could simply check if optimizer is an instance of McoreDistributedOptimizer and return False knowing that MCoreDistOpt mean the checkpoint was produced with at least 24.07 NeMo version so there will be no mismatch?

With the `fully_parallel_model_space` MCoreDistOpt format, the param_groups will be at different location than  `['optimizer'].data['param_groups']`

Reply:
But I'm ok with the current version too